src/main/scala/mesosphere/marathon/core/instance/update/InstanceChangedEventsGenerator.scala

```diff
@@ -52,4 +52,11 @@ object InstanceChangedEventsGenerator {
       taskEvent +: instanceEvent
     }
   }
+
+  def goalChange(instance: Instance): MarathonEvent = InstanceGoalChanged(
```

Comment:
Why not create an alternative constructor for `InstanceGoalChanged`?

Reply:
From my perspective, it seems like this rather is the alternative constructor, but is here instead of in a companion object

Reply:
There are no usages of this anyway so I dropped it ✅